src/samplers/probabilistic_sampler.js

```diff
@@ -11,12 +11,14 @@
 // or implied. See the License for the specific language governing permissions and limitations under
 // the License.
 
-import * as constants from '../constants.js';
+import * as constants from '../constants';
+import LegacySamplerV1Base from './_adapt_sampler';
 
-export default class ProbabilisticSampler implements LegacySamplerV1 {
+export default class ProbabilisticSampler extends LegacySamplerV1Base implements LegacySamplerV1 {
   _samplingRate: number;
 
   constructor(samplingRate: number) {
+    super('ProbabilisticSampler');
```

Comment:
Is calling by name the preferred way of invoking super? It seems pretty fragile

Reply:
it simply passes the "name" that is used in toString and uniqueName() functions.